### ISSUE TITLE
Infinite loop and iframe creation when renewing token (RENEW_TOKEN) with Adal.js 1.0.10

### ISSUE THREAD

**Issue by fgrmt** [AzureAD/azure-activedirectory-library-for-js#298]
I'm currently facing an authentication issue when coming back to my Angular.js w/ Adal.js web app after a while (~ a few hours).
I open a web browser, enter the web app URL and it triggers a `RENEW_TOKEN` infinite loop.

**See attached console logs using VERBOSE mode:**
[Adal.js.1.0.10.renew-token.issue.txt](https://github.com/AzureAD/azure-activedirectory-library-for-js/files/303430/Adal.js.1.0.10.renew-token.issue.txt)

I also noticed that clearing cookies/websites data solves temporarily the issue by triggering the authentication process again. (I guess it solves the issue until the AAD cookie expires again)

**ADAL.JS version used: 1.0.10**

### routeProvider configuration:

```
    $routeProvider
    .when('/', {
        templateUrl: 'views/homePage.html',
        requireADLogin: true,
    })
    .when('/home/', {
        templateUrl: 'views/homePage.html',
        requireADLogin: true,
    })
    .otherwise({
        redirectTo: '/'
    });
```
### adalProvider:
```
    adalProvider.init({
        instance: 'https://login.microsoftonline.com/',
        tenant: '*MyTenant*.onmicrosoft.com',
        clientId: '*MyWebAppClientID*',
        extraQueryParameter: 'nux=1',
        endpoints: { /*.........*/ },
	  cacheLocation: 'localStorage' }, $httpProvider);
```

**Comment by tushargupta51** [comment]
@fgrmt Can you try using the files here: https://github.com/AzureAD/azure-activedirectory-library-for-js/tree/release1.0.11/lib and see if you still run into the loop issue? Note that these are not release files, they will be official released with 1.0.11

**Comment by fairmutex** [comment]
Yep Still happens

**Comment by jeromeyackley** [comment]
We are having the same issue.

**Comment by fgrmt** [comment]
@tushargupta51 Thanks for your response. As suggested, I tried using 1.0.11 but unfortunately I'm still facing the issue.

With version 1.0.11:
* **Using IE**: the error shows up in the console and execution seems to stop without going into an infinite loop. As a result, the web app is never loaded (white page). 
  **See attached console logs (VERBOSE): [AdalJS-1.0.11-logs-IE.txt](https://github.com/AzureAD/azure-activedirectory-library-for-js/files/306066/AdalJS-1.0.11-logs-IE.txt)**
  BTW, both `login.microsoftonline.com` and the web app URL are in the Trusted Sites list in IE.

* **Using Chrome**: the `RENEW_TOKEN` goes into an infinite loop like with version 1.0.10.
  **See attached console logs (VERBOSE): [AdalJS-1.0.11-logs-Chrome.txt](https://github.com/AzureAD/azure-activedirectory-library-for-js/files/306067/AdalJS-1.0.11-logs-Chrome.txt)**

**Comment by tushargupta51** [comment]
@fgrmt Something is triggering "views/homePage.html" to load again and again. In release1.0.11, if token renewal fails, we stop the redirect but the default page gets loaded in iframe. Maybe that is causing these infinite page loads. Do you see nested iframe creation?

**Comment by AOlkar** [comment]
We haven't tried with 1.0.11 yet, but with 1.0.10 we see the same issue. We observe that initial login causes the app to cycle a few times, as well as when token expires and needs to be renewed it goes into the infinite loop with nested iframes.

**Comment by fgrmt** [comment]
@tushargupta51 Thanks for the clarification. I really appreciate.
Indeed, we see nested iframe creation with the infinite loop.

Today, I also had a chance to do more testing regarding this issue:
* I downgraded ADAL.JS to 1.0.7 and then our web app is working again then.
* Upgrading again to 1.0.8 or newer 
Consequently, it looks like this issue is related to a change introduced in the 1.0.8 version.

**Comment by tushargupta51** [comment]
@AOlkar @fgrmt Nested iframe is a known issue. We have a potential solution for this, providing a different html that will be loaded in iframe instead of app's root page (that requires authentication in some cases).

I will update this issue with more details once the fix is available in dev branch, so that you can verify it addresses your issue as well.

**Comment by ruisilva450** [comment]
Any ETA for this issue?
I want to use this library for production code and this issue the main blocker. I tried to solve but since you seem to have a potential solution...
Thanks

**Comment by vukdavid** [comment]
We are facing the same issue. Any ETA please?

**Comment by tushargupta51** [comment]
@ruisilva450 @fgrmt @vukdavid Can you guys try one of the solutions listed here: https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/FAQs#q1-my-app-is-re-loading-every-time-adal-renews-a-token. Please use the bits in dev branch to test if it resolves your issue. Current plan is to release 1.0.11 in second week of july.

**Comment by tushargupta51** [comment]
We have released 1.0.11. Please use those bits and have a look here: https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/FAQs#q1-my-app-is-re-loading-every-time-adal-renews-a-token. Let me know if you still run into any issues.
